plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -477,10 +480,22 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             if( future.valid() ) {
                future.wait();
                app().post( priority::low, [self, future{std::move(future)}, persist_until_expired, next{std::move( next )}, trx_id]() mutable {
-                  auto exception_handler = [&next, trx_id](fc::exception_ptr ex) {
+                  auto exception_handler = [self, &next, trx_id](fc::exception_ptr ex) {
                     fc_dlog(_trx_failed_trace_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${txid} : ${why} ",
                             ("txid", trx_id)("why",ex->what()));
                      next(ex);
+
+                    if (_trx_trace_dump_log.is_enabled(fc::log_level::debug)) {
+                        if (self->_unapplied_transactions.get_trx(trx_id)) {
+                            //Dump transaction
+                            std::shared_ptr<transaction> transaction_ptr = std::make_shared<transaction>(self->_unapplied_transactions.get_trx(trx_id)->packed_trx()->get_transaction());
+                            auto entire_trx = self->chain_plug->get_entire_trx_trace(transaction_ptr);
+                            fc_dlog(_trx_trace_dump_log, "[TRX_TRACE] [Speculative execution is REJECTING tx: ${txid} : ${why}]: \n${entire_trx} ",
+                                    ("txid", trx_id)
+                                    ("why",ex->what())
+                                    ("entire_trx", fc::json::to_pretty_string(entire_trx)));
```

Comment:
I think it is worth asking if file support is needed here. Having all this logging mixed in with the rest of the logging seems like it would make it rather unusable. I can't imagine wanting to_pretty_string unless this is going to a file.

Reply:
Also the `\n` in the logging also messes with "normal" logging which is one statement per line.

Reply:
Ok. I agree.  I will delete all the call `to_pretty_string()` and `\n`. I though it would be more helpful to look at formatted transaction traces in the log while debugging.

Reply:
Fixed